pkg/etcdutil/etcdutil.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2016 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package etcdutil
+
+import (
+	"net/url"
+
+	"github.com/coreos/etcd/etcdserver"
+	"github.com/coreos/etcd/pkg/types"
+	"github.com/juju/errors"
+	"github.com/ngaut/log"
+	"github.com/pingcap/pd/pkg/apiutil"
+	"github.com/pingcap/pd/pkg/testutil"
+)
+
+// CheckClusterID checks Etcd's cluster ID, returns an error if mismatch.
+// This function will never block even quorum is not satisfied.
+func CheckClusterID(localClusterID types.ID, peerURLs []string) error {
+	if len(peerURLs) == 0 {
+		return nil
+	}
+
+	var remoteClusterID types.ID
+	for _, u := range peerURLs {
+		u, gerr := url.Parse(u)
+		if gerr != nil {
+			return errors.Trace(gerr)
+		}
+		trp := apiutil.NewHTTPTransport(u.Scheme)
+
+		// For tests, unix RoundTripper and change scheme to http.
+		for i := range peerURLs {
+			peerURLs[i] = testutil.UnixToHTTP.Replace(peerURLs[i])
```

Comment:
don't let common code call test function. you can put UnixToHTTP here and add an explanation.